regions/core/pixcoord.py

```diff
@@ -33,6 +33,8 @@ class PixCoord(object):
     """
 
     def __init__(self, x, y):
+        if np.asanyarray(x).shape != np.asanyarray(y).shape:
```

Comment:
This is allowed in SkyCoord and ideally we'd behave the same way in PixCoord:
```
>>> from astropy.coordinates import SkyCoord
>>> SkyCoord(42, [1, 2, 3], unit='deg')
<SkyCoord (ICRS): (ra, dec) in deg
    [(42., 1.), (42., 2.), (42., 3.)]>
```
Also, the way you've written the check, to create arrays in init just to check a shape is a bit inefficient, and the error message isn't great (not mentioning the shape) but printing out the full x and y array which can be huge depending on what Numpy print options the user has set. Again, ideally we should be as similar to SkyCoord as possible:
```
>>> SkyCoord([42, 43], [1, 2, 3], unit='deg')

ValueError: Input parameters lon and lat cannot be broadcast
```
Not sure what to do here.
@sushobhana - Maybe remove the check for now and open a new reminder issue to add tests / checking for PixCoord to behave like SkyCoord?
If it's not complicated to do (I have no idea), and if you have time, you could also attempt it in this PR.
But I'm -1 on putting in this check as-is; it could even break existing users.

Reply:
The first case that you mentioned will be easy to deal with.

The second case with the ValueError message can be modified to mention only the dimensions and not the values of x and y.

If these two things are only concern , I think it can be resolved in this PR itself.
Or else if you feel there is more to do , we can create a separate issue to deal with this later.

Reply:
@sushobhana - Then please try to resolve this in this PR. Let me know when you made a change and I should have another look.